hw/drivers/sensors/bma253/src/bma253.c

```diff
@@ -795,10 +1004,34 @@ bma253_set_power_settings(const struct bma253 * bma253,
     if (rc != 0) {
         return rc;
     }
+
+
+    if (BMA253_POWER_MODE_NORMAL == power_settings->power_mode) {
+        rc = bma253_clear_fifo(bma253);
+        BMA253_DRV_CHECK_RC(rc);
+    }
+
     rc = set_register((struct bma253 *)bma253, REG_ADDR_PMU_LPW, data[0]);
     if (rc != 0) {
         return rc;
     }
+    ((struct bma253 *)bma253)->power = power_settings->power_mode;
+
+    //immediately after changing power mode, a delay is needed
```

Comment:
Please convert all C++ comments to C comments

Reply:
changed